cli/config/config_test.go

```diff
@@ -552,13 +552,73 @@ func TestLoadDefaultConfigFile(t *testing.T) {
 func TestConfigPath(t *testing.T) {
 	oldDir := Dir()
 
-	SetDir("dummy1")
-	f1 := Path("a", "b")
-	assert.Equal(t, f1, filepath.Join("dummy1", "a", "b"))
-
-	SetDir("dummy2")
-	f2 := Path("c", "d")
-	assert.Equal(t, f2, filepath.Join("dummy2", "c", "d"))
+	for _, tc := range []struct {
+		name        string
+		path        []string
+		dir         string
+		expected    string
+		expectedErr string
+	}{
+		{
+			name:     "valid_path",
+			path:     []string{"a", "b"},
+			dir:      "root_dir",
+			expected: filepath.Join("root_dir", "a", "b"),
+		},
+		{
+			name:     "valid_path_dir_is_absolute_unix",
+			path:     []string{"a", "b"},
+			dir:      "/root_dir",
+			expected: filepath.Join("/root_dir", "a", "b"),
+		},
+		{
+			name:     "valid_path_dir_is_absolute_windows",
+			path:     []string{"a", "b"},
+			dir:      "C:/root_dir",
+			expected: filepath.Join("C:/root_dir", "a", "b"),
+		},
+		{
+			name:        "invalid_relative_path",
+			path:        []string{"a", "..", "..", "b"},
+			dir:         "root_dir",
+			expectedErr: `is outside of root config directory "root_dir"`,
+		},
+		{
+			name:        "invalid_relative_path_dir_is_absolute_unix",
+			path:        []string{"a", "..", "..", "b"},
+			dir:         "/root_dir",
+			expectedErr: `is outside of root config directory "/root_dir"`,
+		},
+		{
+			name:        "invalid_relative_path_dir_is_absolute_windows",
+			path:        []string{"a", "..", "..", "b"},
+			dir:         "C:/root_dir",
+			expectedErr: `is outside of root config directory "C:/root_dir"`,
+		},
+		{
+			name:        "invalid_absolute_path_unix",
+			path:        []string{"/a", "..", ".."},
+			dir:         "root_dir",
+			expectedErr: `is outside of root config directory "root_dir"`,
+		},
+		{
+			name:        "invalid_absolute_path_windows",
+			path:        []string{"C:/", "..", ".."},
```

Comment:
Does this test DTRT when `GOOS=linux` (or `!=windows` generally)?

Reply:
TIL [DTRT](https://www.urbandictionary.com/define.php?term=DTRT) - had to look that one up 😅

Reply:
If we're testing for these situations, we also might need a test for 

- `C:\` (backslash)
- Embedded travering (for better name);  `[]string{"a", "../../..", "b"}` (although we might enter the area of testing Go stdlib itself)

Reply:
The reason I ask is that `path/filepath` is (by definition) platform specific, so on Unix-y platforms it doesn't treat e.g .`C:/` as special (I expect it thinks it is a directory in the root called `C:`).

Reply:
Yeah, on second look this won't work as the build is on unix. Probably easiest to simply remove the "windows" tests for now